dev-3/dev-3/Transliterator.cs

```diff
@@ -0,0 +1,92 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace dev_3
+{
+    /// <summary>
+    /// Class for transliting sentences.
+    /// Fields may look strange, because methods are working with indices of arrays.
+    /// All transliterations are made according to transliteration table at https://www.westernunion.ru/ru/ru/transliteration-table.html
+    /// </summary>
+    public class Transliterator
+    {
+        public string Sentence { get; }
+        private bool IsRussian = false;
+        private string[] _translitRule = new string[] {"a","b", "v", "g", "d", "e", "zh", "z", "i", "y", "k", "l",
+        "m", "n", "o", "p", "r", "s", "t", "u", "f", "kh", "ts", "ch", "sh", "sch", "", "y", "", "e", "yu", "ya","","yo"};
+        private string[] _twoSymbolsTranslit = new string[] {"","", "", "", "", "", "zh", "", "", "", "", "",
+        "", "", "", "", "", "", "", "", "", "kh", "ts", "ch", "sh", "sch", "", "", "", "", "yu", "ya","","yo"};
+        public const int RUSSIANLOWERCASEA = 1072;
+        public const int LATINLOWERCASEA = 97;
+        public IEnumerable<int> LatinLowerCaseRange = Enumerable.Range(97, 27);
+        public IEnumerable<int> RussianLowerCaseRange = Enumerable.Range(1072, 34);
+        /// <summary>
+        /// Constructor that takes string, checks its validity and sets mode of transliteration
+        /// </summary>
+        /// <param name="sentence">this string will be processed</param>
+        public Transliterator(string sentence)
+        {
+            CheckValidity(sentence);
+            Sentence = sentence.ToLower();
+            SetMode();
+        }
+        /// <summary>
+        /// There are 2 modes : from russian to translit and from translit to russian
+        /// </summary>
```

Comment:
Not SetMode, set means that you pass argument to method and set private value, better name DetermineSourceLanguage